tests/performance/utils.js

```diff
@@ -21,13 +21,58 @@ function median( array ) {
  * @return {string} Result file name.
  */
 function getResultsFilename( fileName ) {
-	const prefixArg = process.argv.find( ( arg ) => arg.startsWith( '--prefix' ) );
-	const fileNamePrefix = prefixArg ? `${prefixArg.split( '=' )[1]}-` : '';
+	const prefixArg = process.argv.find( ( arg ) =>
+		arg.startsWith( '--prefix' )
+	);
+	const fileNamePrefix = prefixArg ? `${ prefixArg.split( '=' )[ 1 ] }-` : '';
 	const resultsFilename = fileNamePrefix + fileName + '.results.json';
 	return resultsFilename;
 }
 
+/**
+ * Returns time to first byte (TTFB) using the Navigation Timing API.
+ *
+ * @see https://web.dev/ttfb/#measure-ttfb-in-javascript
+ *
+ * @return {Promise<number>}
+ */
+async function getTimeToFirstByte() {
+	return page.evaluate( () => {
+		const { responseStart, startTime } =
+			performance.getEntriesByType( 'navigation' )[ 0 ];
+		return responseStart - startTime;
+	} );
+}
+
+/**
+ * Returns the Largest Contentful Paint (LCP) value using the dedicated API.
+ *
+ * @see https://w3c.github.io/largest-contentful-paint/
+ * @see https://web.dev/lcp/#measure-lcp-in-javascript
+ *
+ * @return {Promise<number>}
+ */
+async function getLargestContentfulPaint() {
+	return page.evaluate(
+		() =>
+			new Promise( ( resolve ) => {
```

Comment:
Is it considered a best practice to incorporate try-catch blocks around asynchronous operations within functions to handle potential exceptions, particularly when working with promises?

Reply:
The test runner is basically one giant try-catch block

Reply:
Ok. thanks for sharing.